lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -111,7 +113,51 @@ export default function FilePickerApp({ onSubmit }) {
    * render.
    */
   const [shouldSubmit, setShouldSubmit] = useState(false);
-  const submit = useCallback(() => setShouldSubmit(true), []);
+
+  const [deeplinkingFields, setDeeplinkingFields] = useState(
+    /** @type {Record<string,string>} */ null
```

Comment:
```suggestion
    /** @type {Record<string,string>|null} */ (null)
```

For technical reasons, the expressions `@type` annotates have to be wrapped in parens. Also the type should include the `null` possibility.